pkg/build/builder/cmd/builder.go

```diff
@@ -106,19 +110,21 @@ func setupSourceSecret(sourceSecretName string, scmAuths []scmauth.SCMAuth) erro
 	}
 
 	if len(scmAuthsPresent) == 0 {
-		return fmt.Errorf("no auth handler was found for the provided secret %q",
+		return nil, fmt.Errorf("no auth handler was found for the provided secret %q",
 			sourceSecretName)
 	}
 
+	url := &url.URL{}
```

Comment:
You can also just declare the variable without allocating a value for it (var u *url.URL)